src/services/document_service.py

```diff
@@ -318,32 +284,21 @@ def update_document(
         logger.info(f"Updated entity '{reference}'")
         return {"data": tree_node_to_dict(node)}
 
-    def remove_by_path(self, data_source_id: str, directory: str, attribute: str = None):
-        if attribute:
-            raise ApplicationException(
-                "Removing a document by path in combination with attribute is not yet supported"
-            )
-        directory = directory.rstrip("/").lstrip("/")
+    def remove(self, reference: str) -> None:
+        data_source_id, _reference = split_data_source_and_reference(reference)
         data_source = self.repository_provider(data_source_id, self.user)
 
-        if "/" in directory:
-            parent = self.get_document(f"/{data_source_id}/{'/'.join(directory.split('/')[0:-1])}", depth=1)
-
-            sub: ResolvedReference = resolve_reference(f"/{data_source_id}/{directory}", self.get_data_source)
-
-            # find the node id of the child with uid equal to child_uid
-            child_node_ids = (child.node_id for child in parent.children[0].children if child.uid == sub.document_id)
-            child_node_id = next(child_node_ids)
-
-            # The first child of a directory is always 'content'
-            parent.children[0].remove_by_child_id(child_node_id)
-            self.save(parent, data_source_id)
-            delete_document(data_source, document_id=sub.document_id)
+        resolved_reference: ResolvedReference = resolve_reference(reference, self.get_data_source)
+        # If the reference goes through a parent, get the parent document
+        document = data_source.get(resolved_reference.document_id)
```

Comment:
Could you move this inside the if sentence? There is no need to get the document if there is no attribute path.